pgx-pg-sys/src/lib.rs

```diff
@@ -445,6 +450,42 @@ mod all_versions {
             context: *mut ::std::os::raw::c_void,
         ) -> bool;
     }
+
+    #[inline]
+    pub fn get_arr_data_ptr<T>(arr: *mut super::ArrayType) -> *mut T {
+        unsafe { pgx_ARR_DATA_PTR(arr) as *mut T }
+    }
```

Comment:
This is not a safe function. It is unsound to call this if `*mut ArrayType` is the null pointer or an invalid pointer, as it involves **reading a field** of the pointee to determine what to do next. A function must **not** be safe if it is memory-unsafe to call it with **any** valid input for that type, **especially** including the null pointer.

This function may be desirable but it **must** be `unsafe`.